File: vm/bench/php/maps.php
<?php

$counts = [];
for ($i = 0; $i < 500000; $i++) {
    $word = 'w'.($i * 31 % 5003);
    $counts[$word] ??= 0;
    $counts[$word]++;
}
$total = 0;
foreach ($counts as $w => $c) {
    $total += $c;
}
echo count($counts).' '.$total.' '.$counts['w17'], "\n";

File: vm/bench/python/maps.py
def main():
    counts = {}
    for i in range(500000):
        word = "w" + str(i * 31 % 5003)
        if word not in counts:
            counts[word] = 0
        counts[word] += 1
    total = 0
    for w, c in counts.items():
        total += c
    print(f"{len(counts)} {total} {counts['w17']}")


main()
